### PR TITLE
Added Contributors window.

### PR DESCRIPTION
**Pull request by nuno**

### CODE REVIEW

File: app/controllers/lunches/details/contributors.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+var loader,
+    LoaderInstance;
+
+/**
+ *  Constructor
+ */
+( function constructor( args ) {
```
Comment (hansemannn) [inline app/controllers/lunches/details/contributors.js]: Something is wrong with your formatting. It put's spaces before and after every bracket. This should confirm with the rest the code

File: app/controllers/lunches/details/contributors.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+function populateContributors() {
+    var api = require( "/api" );
+    loader.show();
+
+    var items = [];
+    api.getContrib( {
+
+    }, function( payload ) {
```
Comment (hansemannn) [inline app/controllers/lunches/details/contributors.js]: Leave out the 1st parameter if there are no params. Adjst the api method accordingly.

File: app/controllers/lunches/details/contributors.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        console.log( payload[ 0 ].login );
```
Comment (hansemannn) [inline app/controllers/lunches/details/contributors.js]: Remove log

File: app/controllers/lunches/details/contributors.js (hunk 2 above)
Comment (hansemannn) [inline app/controllers/lunches/details/contributors.js]: Rename `payload` to `e` to match the code-style.

File: app/controllers/settings/index.js [hunk 4]
```diff
@@ -120,6 +120,10 @@ function openAbout() {
     nav.openWindow(Alloy.createController("/settings/webview", "about").getView());
 }
 
+function openContributors(e){
+  nav.openWindow(Alloy.createController("/lunches/details/contributors", "contributors").getView());
```
Comment (hansemannn) [inline app/controllers/settings/index.js:125]: Please use 4-space indentation and fix it for Android (see other classes for common `open` examples)

File: app/i18n/de/strings.xml [hunk 5]
```diff
@@ -102,5 +102,6 @@
 	<string name="upload_first_photo_now">Lade jetzt das erste Foto hoch ...</string>
 	<string name="up_or_down">👍 oder 👎</string>
 	<string name="price">Preis</string>
+	<string name="contributors">Contributors</string>
```
Comment (hansemannn) [inline app/i18n/de/strings.xml]: `Mitwirkende`

File: app/i18n/es/strings.xml [hunk 6]
```diff
@@ -102,5 +102,6 @@
 	<string name="upload_first_photo_now">Sube la primera foto ahora ...</string>
 	<string name="up_or_down">👍 o 👎</string>
 	<string name="price">Precio</string>
+	<string name="contributors">Contributors</string>
```
Comment (hansemannn) [inline app/i18n/es/strings.xml]: You will know

File: app/i18n/pt/strings.xml [hunk 7]
```diff
@@ -100,5 +100,6 @@
 	<string name="upload_first_photo_now">Carregar a primeira foto agora ...</string>
 	<string name="up_or_down">👍 ou 👎</string>
 	<string name="price">Preço</string>
+	<string name="contributors">Contributors</string>
```
Comment (hansemannn) [inline app/i18n/pt/strings.xml]: You will know

File: app/lib/api.js [hunk 8]
```diff
@@ -13,6 +13,23 @@ function performFallback(cb) {
 }
 
 /**
+ * Faillback for contributors
```
Comment (hansemannn) [inline app/lib/api.js]: Typo, it's `Fallback`

File: app/lib/api.js [hunk 9]
```diff
@@ -13,6 +13,23 @@ function performFallback(cb) {
...
+ */
+function contribFallback(cb) {
+	var dummyLunches = Ti.Filesystem.getFile(Ti.Filesystem.getResourcesDirectory(), 'json/contributors.json');
```
Comment (hansemannn) [inline app/lib/api.js]: contributors are no lunches

File: app/lib/api.js [hunk 10]
```diff
@@ -109,3 +126,32 @@ exports.getLunches = function(params, cb) {
 		return;
 	}
 };
+
+/**
+ * Get all contributors
+ *	@param {Object} params The GET parameter to form the URL.
+ *	@param {Callback} cb The callback to be invoked after the asyncronous request.
+ *	@return void
```
Comment (hansemannn) [inline app/lib/api.js]: Either the whole comment tabbed or spaced.

File: app/lib/api.js [hunk 11]
```diff
@@ -109,3 +126,32 @@ exports.getLunches = function(params, cb) {
...
+			},
+			error : function() {
+				cb({success: false});
+			}
+		});
+		request.load();
+	} catch(ex) {
+		contribFallback(cb);
+		console.warn(ex);
+		return;
```
Comment (hansemannn) [inline app/lib/api.js]: No `return` necessary, there is nothing else to go to

File: app/lib/request.js [hunk 12]
```diff
@@ -164,7 +164,8 @@ var Request = function(_args) {
 
 		c.open(reqType, url);
 		
-		c.setRequestHeader('Authorization', 'Basic ' + getCredentials());
+		//getCredentials() call "auth"lib and because at this moment this lib not exists some network call faill, it is comment out for now.
+		//c.setRequestHeader('Authorization', 'Basic ' + getCredentials());
```
Comment (hansemannn) [inline app/lib/request.js]: Can't do that, I need that for the prod API. Better use `if (!args.external)` and wrap it in that.

File: app/styles/lunches/details/contributors.tss [hunk 13]
```diff
@@ -0,0 +1,34 @@
+"#list": {
+	style: Ti.UI.iOS.ListViewStyle.GROUPED,
+	listSeparatorInsets: {
+		left: 15
+	}
+}
+
+"#list[platform=ios]": {
```
Comment (hansemannn) [inline app/styles/lunches/details/contributors.tss:1]: Move contents from `#list` to this one, those are iOS-only as well

File: app/styles/lunches/details/contributors.tss [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+	top: -5
+}
+
+".title": {
+	left: 60,
+	font: {
+		fontSize: 17 // TODO: This is the default for iOS, might check Android manually
```
Comment (hansemannn) [inline app/styles/lunches/details/contributors.tss]: See other Android changes I made to handle this for parity

File: app/styles/lunches/details/contributors.tss [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+	}
+}
+
+".avatar": {
+  width: 80,
+  left: -18
```
Comment (hansemannn) [inline app/styles/lunches/details/contributors.tss]: Indentation messed up

File: app/styles/lunches/details/contributors.tss [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+".notSelectable": {
+	selectionStyle: Ti.UI.iOS.ListViewCellSelectionStyle.NONE
```
Comment (hansemannn) [inline app/styles/lunches/details/contributors.tss:21]: Wrap for iOS-only

File: app/styles/lunches/details/contributors.tss [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+"ListItem": {
+  selectionStyle: Ti.UI.iOS.ListViewCellSelectionStyle.NONE,
+	listSeparatorInsets: {
+		left: 0
+	},
+  height: 80
```
Comment (hansemannn) [inline app/styles/lunches/details/contributors.tss]: Same as above, wrap properly.

File: app/views/lunches/details/contributors.xml [hunk 18]
```diff
@@ -0,0 +1,13 @@
+<Alloy>
+	<Window id="window" titleid="contributors" onOpen="populateContributors">
+	<ListView id="list" defaultItemTemplate="contrib">
+		<Templates>
+			<ItemTemplate name="contrib" class="notSelectable">
+				<ImageView class="avatar" bindId="avatar" />
+				<Label class="title" bindId="title" />
+			</ItemTemplate>
+		</Templates>
+		<ListSection id="section" headerTitle="L('contributors')" />
```
Comment (hansemannn) [inline app/views/lunches/details/contributors.xml]: No need for another title, the window title should be fine.

**Review by hansemannn — CHANGES_REQUESTED** [review]

File: app/lib/api.js (hunk 11 above)
Comment (nuno) [reply app/lib/api.js]: You return in all the others, that is way I put there ;)

File: app/controllers/settings/index.js (hunk 4 above)
Comment (nuno) [reply app/controllers/settings/index.js:125]: What is the difference between mine and openAbout()  ?

Comment (hansemannn) [reply app/controllers/settings/index.js:125]: Different indentation. And `openAbout` needs to be fixed as well. I'm not perfect :-)

File: app/lib/api.js (hunk 11 above)
Comment (hansemannn) [reply app/lib/api.js]: Remove them all then :-)

**Review by hansemannn — APPROVED** [review]

**Comment by hansemannn** [comment]
Whoops, just noticed that more is wrong. 

- Why is it placed in /lunches/details/contributors?
- Why did you pass the "contributors" arg? It's only used for the webview to locate the resource

Will fix it quickly locally 👍

**Comment by nuno** [comment]
Ops, I guess Im not perfect ;) also for a little time I had to read how the code/api work, is not bad...

**Comment by hansemannn** [comment]
Man, it's great :-) So happy to have it in! Just pushed a quick fix to match the design and include the  number of contributions as well 🚀

**Pull request merged**